src/components/transactions/TxDetails/TxData/DecodedData/MethodCall.tsx

```diff
@@ -0,0 +1,55 @@
+import NamedAddressInfo from '@/components/common/NamedAddressInfo'
+import { Box, Typography } from '@mui/material'
+
+const MethodCall = ({
+  method,
+  contractAddress,
+  contractName,
+  contractLogo,
+}: {
+  method: string
+  contractAddress: string
+  contractName?: string
+  contractLogo?: string
+}) => {
+  return (
+    <>
+      <Typography fontWeight="bold" display="flex" alignItems="center" gap=".5em" pb={1.5} component="div">
+        Call{' '}
+        <Typography
+          component="code"
+          variant="body2"
+          sx={{
+            backgroundColor: 'background.main',
+            px: 1,
+            py: 0.5,
+            borderRadius: 0.5,
+            fontFamily: 'monospace',
+            whiteSpace: 'nowrap',
+          }}
+        >
+          {method}
+        </Typography>{' '}
+        on
+        <NamedAddressInfo
+          address={contractAddress}
+          name={contractName}
+          customAvatar={contractLogo}
+          onlyName
+          hasExplorer
+          showCopyButton
+          avatarSize={26}
+        />
```

Comment:
What do you think about using this component in `generateDataRowValue` if there is a name for an address to make it better fit with the rest of the layout?

Current:
<img width="619" alt="Screenshot 2024-07-22 at 12 14 59" src="https://github.com/user-attachments/assets/9f4f57f6-53ab-481c-97d2-9b04102202be">

With NamedAddressInfo:
<img width="601" alt="Screenshot 2024-07-22 at 12 14 35" src="https://github.com/user-attachments/assets/cdc912f9-7f16-41de-8a02-4672e872bbc8">

Reply:
I think this section _must_ display the address. In fact I would make it long.